celeryutils/persist_on_failure.py

```diff
@@ -0,0 +1,52 @@
+"""
+Celery utility code for persistent tasks.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from celery import Task
+
+from .models import FailedTask
+
+
+# pylint: disable=abstract-method
+class PersistOnFailureTask(Task):
+    """
+    Custom Celery Task base class that persists task data on failure.
+    """
+
+    abstract = True
+    typing = False
```

Comment:
Yes.  There's a test that calls the task with the wrong args, and without this flag it fails under celery 4.